### DISCUSSION TITLE
[Q][jdk 17(Component)] dubbo reactor

### DISCUSSION THREAD

**Discussion by 907474646** [apache/dubbo#14771] [Question] [answered]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

Why, when I used the Dubbo reactor mode in the provider, did I find that Dubbo is still running in the Dubbo handler thread pool? Under high concurrency, it still reports errors indicating that the thread pool is full.
I user dubbo 3.2.14 ，jdk 17

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
What thread pool do you want?

**Reply by 907474646** [reply]
I use reactor mode，why it does not been run in reactor thread .I mean it should don't be  occured the error with thread pool full,beacuse 
 in reactor mode,one request is only one event;

**Reply by 907474646** [reply]
This is error detail,
|o.a.d.r.p.t.s.TripleServerStream:| [DUBBO] submit onData task failed, dubbo version: 3.2.14, current host: 10.70.34.93, error code: 4-11. This may be caused by , go to https://dubbo.apache.org/faq/4/11 to find instructions. java.util.concurrent.RejectedExecutionException: Thread pool is EXHAUSTED! Thread Name: DubboServerHandler-10.70.34.93:28083, Pool Size: 50 (active: 50, core: 50, max: 50, largest: 50), Task: 5186 (completed: 5136), Executor status:(isShutdown:false, isTerminated:false, isTerminating:false), in tri://12.75.34.53:3083! ~~       at org.apache.dubbo.common.threadpool.support.AbortPolicyWithReport.rejectedExecution(AbortPolicyWithReport.java:128) ~~   at java.base/java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:833)

**Reply by icodening** [reply, marked as answer]
>  client send request ->  server -> parse request -> submit threadpool -> adapting reactor invoke

In triple, after receiving the request, it will try to submit it to the thread pool for execution, and handle the adaptation of the reactor in the thread pool.
